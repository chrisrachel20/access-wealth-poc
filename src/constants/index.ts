export const  NET_QTY = 'Net Quantity';
export const  MARKET_PRICE = 'Market Price';
export const  MARKET_VALUE = 'Market Value';
export const PORTFOLIO_VALUE = 'Portfolio Valuation';
export const INVESTOR_NAME = 'Investor Name:';
export const INVESTOR_ACCOUNT_CURRENCY = 'Investor Account Currency:';
export const VALUATION_CURRENCY = 'Valuation Currency:';
export const INVESTOR_PORTFOLIO_VALUE = 'Portfolio Value:';
export const ACCESS_WEALTH = "Access Wealth";
export const SIDEBAR = [
  {
    title: "Dashboard",
    route: "/dashboard",
  },
  {
    title: "Transactions",
    route: "/transactions"
  },
];
export const PIE_COLORS = ["#0088FE", "#00C49F", "#FFBB28", "#FF8042"];
export const LINE_COLORS = ["#0d6efd", "#6f42c1", "#20c997", "#0dcaf0", "#198754"];