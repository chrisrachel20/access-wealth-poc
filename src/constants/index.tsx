import DashboardIcon from "@mui/icons-material/Dashboard";
import LineAxisIcon from "@mui/icons-material/LineAxis";
import LogoutIcon from "@mui/icons-material/Logout";

export const SIGN_IN = " Sign In";
export const LOGIN_ERROR = "Unauthorized User. Please enter valid credentials";
export const NET_QTY = "Net Quantity";
export const MARKET_PRICE = "Market Price";
export const MARKET_VALUE = "Market Value";
export const PORTFOLIO_VALUE = "Portfolio Valuation";
export const INVESTOR_NAME = "Investor Name";
export const INVESTOR_ACCOUNT_CURRENCY = "Investor Account Currency:";
export const VALUATION_CURRENCY = "Valuation Currency:";
export const INVESTOR_PORTFOLIO_VALUE = "Portfolio Value";
export const ACCESS_WEALTH = "Access Wealth";
export const SIDEBAR = [
  {
    title: "Dashboard",
    route: "/dashboard",
    icon: <DashboardIcon />,
  },
  {
    title: "Transactions",
    route: "/transactions",
    icon: <LineAxisIcon />,
  },
  {
    title: "Sign Out",
    route: "/login",
    icon: <LogoutIcon />,
  },
];
export const PIE_COLORS = [ "#58508d", "#bc5090", "#ff6361", "#ffa600"];
export const LINE_COLORS = [
  "#e71069",
  "#4ec1dd",
  "#2fb898",
  "#852b8e",
  "#e57c25",
];
export const CURRENT_STOCKS = "Purchased Stocks";
