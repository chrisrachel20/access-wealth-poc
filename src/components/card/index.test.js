import { render, screen } from "@testing-library/react";
import { PortfolioCard } from "./index";
import * as Strings from "../../constants/index";
import { useSelector } from "react-redux";

const masterData = {
  portfolio: {
    name: "Test Investor",
    investor: {
      currencyCode: 'USD'
    }
  },
  currencyCode: "USD",
  marketValue: 10000,
};
const props = { masterData };

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(() => mockDispatch),
  useSelector: jest.fn(),
}));

describe("PortfolioCard", () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => {
      return callback(masterData);
    });
  });

  it("Should render component properly", () => {
    const { container } = render(<PortfolioCard {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render the investor's name", () => {
    render(<PortfolioCard {...props} />);
    expect(screen.getByText(Strings.INVESTOR_NAME)).toBeTruthy();
    expect(screen.getByText(masterData.portfolio.name)).toBeInTheDocument();
  });

  it("should render the investor's account currency", () => {
    render(<PortfolioCard {...props} />);
    expect(screen.getByText(Strings.INVESTOR_ACCOUNT_CURRENCY)).toBeTruthy();
  });

  it("should render the valuation currency", () => {
    render(<PortfolioCard {...props} />);
    expect(screen.getByText(Strings.VALUATION_CURRENCY)).toBeTruthy();
  });

  it("should render the investor's portfolio value", () => {
   render(<PortfolioCard {...props} />);
    expect(screen.getByText(Strings.INVESTOR_PORTFOLIO_VALUE)).toBeTruthy();
    expect(screen.getAllByText(masterData.marketValue)).toBeInTheDocument();
  });
});
