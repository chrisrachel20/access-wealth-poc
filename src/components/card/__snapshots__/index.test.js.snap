// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioCard Should render component properly 1`] = `
<div>
  <div
    class="MuiBox-root css-ppgd76"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-wymnel-MuiContainer-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1kt1ewx-MuiTypography-root"
      >
        Portfolio Valuation
      </h4>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-14486nb-MuiContainer-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vmcu6f-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-fc1rjq-MuiContainer-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-12l6q2q-MuiTypography-root"
            >
              Investor Name:
            </h5>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1vbq3ux-MuiTypography-root"
            >
              Test Investor
            </h5>
          </div>
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-fc1rjq-MuiContainer-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-12l6q2q-MuiTypography-root"
            >
              Investor Account Currency:
            </h5>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1vbq3ux-MuiTypography-root"
            >
              USD
            </h5>
          </div>
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-fc1rjq-MuiContainer-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-12l6q2q-MuiTypography-root"
            >
              Valuation Currency:
            </h5>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1vbq3ux-MuiTypography-root"
            >
              USD
            </h5>
          </div>
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-fc1rjq-MuiContainer-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-12l6q2q-MuiTypography-root"
            >
              Portfolio Value:
            </h5>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1vbq3ux-MuiTypography-root"
            >
              10000.00
            </h5>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-6rhd0e-MuiPaper-root-MuiCard-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ovcsdv-MuiTypography-root"
          >
            Available Stocks
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
